### DISCUSSION TITLE
Arabic: list index out of range

### DISCUSSION THREAD

**Discussion by matteoPiccoli** [PaddlePaddle/PaddleOCR#7746] [Q&A]
Hello everyone!

My testing dataset is composed of nearly 60 images, and I'm having troubles analyzing the following two images: 

![0015](https://user-images.githubusercontent.com/95410734/192503063-50a1130b-d202-44f1-9571-830c2f5062ce.png)
![0049](https://user-images.githubusercontent.com/95410734/192503094-4849f79f-46cc-40cd-83ec-128a7b6c370c.png)

I think that the problem could be the recognition of these **characters**:
1. **:**
2. **'**

More in details, text detection works perfectly but when it comes the recognition stage the following error shows up:

```
Traceback (most recent call last):
  File "c:\Users\matte\OneDrive\Documenti\GitHub\ATB-Text-Recognition\atbocr\engine\predictions.py", line 44, in paddle_recognition
    result = Paddle.ocr(cropped_image, cls=False, det=False, rec=True)
  File "C:\Python\Python39\lib\site-packages\paddleocr\paddleocr.py", line 538, in ocr
    rec_res, elapse = self.text_recognizer(img)
  File "C:\Python\Python39\lib\site-packages\paddleocr\tools\infer\predict_rec.py", line 532, in __call__
    rec_result = self.postprocess_op(preds)
  File "C:\Python\Python39\lib\site-packages\paddleocr\ppocr\postprocess\rec_postprocess.py", line 120, in __call__
    text = self.decode(preds_idx, preds_prob, is_remove_duplicate=True)
  File "C:\Python\Python39\lib\site-packages\paddleocr\ppocr\postprocess\rec_postprocess.py", line 82, in decode
    char_list = [
  File "C:\Python\Python39\lib\site-packages\paddleocr\ppocr\postprocess\rec_postprocess.py", line 83, in <listcomp>
    self.character[text_id]
IndexError: list index out of range
```

The Paddle model is initialized as follows:
```
PaddleOCR(use_angle_cls=True, lang="ar", use_gpu=True)
```

Thank you

**Comment by tink2123** [comment]
1. The  `: '` character cannot be recognized because there is a lack of similar data in the training data. If necessary, you can refer to the [training document](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/recognition_en.md#TRAINING) and use your own data finetune.

2. What version of paddleocr is installed? You can try to update to  `paddleocr==2.6.0.1` . The new version of the whl package should fix the IndexError.

**Reply by matteoPiccoli** [reply]
I already installed the 2.6.0.1 version of paddleocr

**Comment by tink2123** [comment]
Can you provide detailed output information for prediction? Let me check whether the model and the dictionary are consistent. Or you can update the Arabic model:

```
wget https://paddleocr.bj.bcebos.com/PP-OCRv3/multilingual/arabic_PP-OCRv3_rec_infer.tar 

tar -xf arabic_PP-OCRv3_rec_infer.tar
```
Modify the recognition model path:

```
PaddleOCR(use_angle_cls=True, lang="ar", use_gpu=True, rec_model_dir="arabic_PP-OCRv3_rec_infer")
```
